src/libtorch.cc

```diff
@@ -1070,52 +1151,41 @@ ModelInstanceState::ProcessRequests(
             &cuda_copy));
   }
 
-  // Request to retrieve all model outputs. 'output_names' and
-  // 'output_tensors' are parallel vectors and so must be kept in
-  // sync.
-  std::vector<const char*> output_names;
-  std::vector<torch::jit::IValue> output_tensors;
-  if (!all_response_failed) {
-    triton::common::TritonJson::Value ios;
-    TRITONSERVER_Error* err =
-        model_state_->ModelConfig().MemberAsArray("output", &ios);
-    if (err == nullptr) {
-      for (size_t i = 0; i < ios.ArraySize(); i++) {
-        triton::common::TritonJson::Value io;
-        err = ios.IndexAsObject(i, &io);
-        if (err != nullptr) {
-          break;
-        }
-
-        // Use names from ModelConfig by reference since the model
-        // config will persist longer than this inference execution.
-        const char* io_name;
-        size_t io_name_len;
-        err = io.MemberAsString("name", &io_name, &io_name_len);
-        if (err != nullptr) {
-          break;
-        }
-
-        output_names.emplace_back(io_name);
-      }
+  // If the instance kind is not GPU, we need to synchronize the CUDA stream
```

Comment:
This is counter-intuitive. For a non-GPU instance, why do we need to synchronize the CUDA stream?

Reply:
The request data can be from GPU and we copy to CPU, need explicit sync as the model execution will not be on the stream used on copy

Reply:
Exactly.